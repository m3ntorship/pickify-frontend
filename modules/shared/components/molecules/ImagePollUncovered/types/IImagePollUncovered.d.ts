declare namespace IImagePollUncovered {
  export interface IProps {
    type: 'circular' | 'vertical';
    percentage: number;
    mostVoted: boolean;
    leastVoted: boolean;
    id: string;
    verticalMeterHeight?: number;
  }
}

export { IImagePollUncovered };
