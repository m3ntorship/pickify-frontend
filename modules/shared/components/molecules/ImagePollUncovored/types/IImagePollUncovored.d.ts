declare namespace IImagePollUncovored {
  export interface IProps {
    type: 'circular' | 'vertical';
    percentage: number;
    mostVoted: boolean;
    leastVoted: boolean;
    id: string;
    verticalMeterHeight?: number;
  }
}

export { IImagePollUncovored };
