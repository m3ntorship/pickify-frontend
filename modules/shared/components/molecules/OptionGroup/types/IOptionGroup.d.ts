declare namespace IOptionGroup {
  export interface IOption {
    id: string;
    value: string;
  }
  export interface IProps {
    id: string;
    groupName?: string;
    // options: { id: string; letter: string; value: string }[];
    // addOptionHandler: (GroupId: string) => void;
    // deleteOptionHandler: (optionId: string) => void;
    // deleteGroupHandler: () => void;
  }
}
export { IOptionGroup };
