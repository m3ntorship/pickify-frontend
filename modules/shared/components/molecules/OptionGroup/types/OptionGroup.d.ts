declare namespace IOptionGroup {
  export interface IOption {
    id: string;
    letter?: string;
    value?: string;
  }
  export interface IProps {
    handleGroupDelete: () => void;
  }
}
export { IOptionGroup };
