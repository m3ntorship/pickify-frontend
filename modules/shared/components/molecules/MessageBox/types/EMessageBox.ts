export enum MessageBoxType {
  Error = 'error',
  Success = 'success',
}
