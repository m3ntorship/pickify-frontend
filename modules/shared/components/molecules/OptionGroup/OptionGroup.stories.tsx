import React from 'react';
import type { Story, Meta } from '@storybook/react';
import OptionGroup from './OptionGroup';
import type { IOptionGroup } from './types/IOptionGroup';

export default {
  title: 'Molecules/OptionGroup',
  component: OptionGroup,
} as Meta;

const Template: Story<IOptionGroup.IProps> = (args) => (
  <OptionGroup {...args} />
);

export const Default = Template.bind({});

Default.args = {
  groupId: '0',
  options: [
    { id: 'test', value: '' },
    { id: 'test1', value: '' },
  ],
};
