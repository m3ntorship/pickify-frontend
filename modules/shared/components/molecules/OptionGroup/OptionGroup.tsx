import React from 'react';
import type { FC, ReactElement } from 'react';
import PlusCircle from '../../icons/plusCircle.svg';
import Option from '../Option/Option';
import type { IOptionGroup } from './types/IOptionGroup';
import * as ETextInput from '../../atoms/TextInputs/types/ETextInput';

const OptionGroup: FC<IOptionGroup.IProps> = (props): ReactElement => {
  const {
    groupId = '1',
    // onChange,
    options,
    setOptions,
    setOptionsInGroup,
    register,
    formSubmitted,
    reset,
    variantMessage,
    textPollState,
    setTextPollState,
    miniSurveyState,
    setMiniSurveyState,
  } = props;
  const randomId = (): string => {
    const randomHelper = 10000000000;
    return `id_${Math.round(Math.random() * randomHelper)}`;
  };

  const initialOptionsLength = 2;
  const initialIndexAdder = 1;
  const optionsLimit = 26;
  const alphabet: string[] = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'.split('');

  const addOptionHandler = (): void => {
    if (setOptionsInGroup) {
      setOptionsInGroup([...options, { id: randomId(), body: '' }], groupId);
    } else if (setOptions) {
      setOptions([...options, { id: randomId(), body: '' }]);
    }
  };
  const deleteOptionHandler = (optionId: string): void => {
    if (setOptionsInGroup) {
      setOptionsInGroup(
        options.filter((option) => option.id !== optionId),
        groupId,
      );
    } else if (setOptions) {
      setOptions(options.filter((option) => option.id !== optionId));
    }
  };

  return (
    <>
      <div className="flex flex-col space-y-2">
        <div className="flex flex-col space-y-2" data-testid="optionsWrapper">
          {options.map((option, index) => (
            <div key={option.id}>
              <Option
                textPollState={textPollState}
                setTextPollState={setTextPollState}
                miniSurveyState={miniSurveyState}
                setMiniSurveyState={setMiniSurveyState}
                id={`${option.id}`}
                letter={alphabet[index]}
                deletable={options.length > initialOptionsLength}
                deleteInputHandler={(): void => {
                  deleteOptionHandler(option.id);
                }}
                placeholder={`Option ${index + initialIndexAdder}`}
                register={
                  register && {
                    ...register(option.id, {
                      required: {
                        value: true,
                        message: 'This field is required',
                      },
                      minLength: { value: 3, message: 'Minimum letters is 3' },
                    }),
                  }
                }
                reset={reset}
                variants={
                  formSubmitted
                    ? variantMessage(option.id)
                    : ETextInput.Variants.Default
                }
              />
              {/* {errors[option.id] && (
                <span className="text-error">{errors[option.id].message}</span>
              )} */}
            </div>
          ))}
        </div>
        {options.length < optionsLimit ? (
          <div className="py-2.5">
            <button
              data-testid="addOptionBtn"
              type="button"
              className="text-accent cursor-pointer flex items-center self-start focus:outline-none"
              onClick={(): void => {
                addOptionHandler();
              }}
            >
              <PlusCircle className="fill-accent mr-1 my-3xxs" />
              <span className="text-sm font-medium">Add choice</span>
            </button>
          </div>
        ) : null}
      </div>
    </>
  );
};

export default OptionGroup;
