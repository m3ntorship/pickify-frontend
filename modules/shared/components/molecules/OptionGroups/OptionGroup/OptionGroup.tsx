import React, { useState } from 'react';
import type { FC, ReactElement } from 'react';
import TextDefault from '../../TextDefault/TextDefault';
import type { IOptionGroup } from './types/IOptionGroup';
import ThreeDotsIcon from '../../../icons/verticalThreeDots.svg';
import CheckSmall from '../../../icons/checkMarkSmall.svg';
import XIcon from '../../../icons/xicon.svg';
import PlusCircle from '../../../icons/plusCircle.svg';

const OptionGroup: FC<IOptionGroup.IProps> = ({
  groupId,
  defaultName,
  deleteGroupHandler,
  addOptionHandler,
  deleteOptionHandler,
  options,
}): ReactElement => {
  const [groupName, setGroupName] = useState<string>(defaultName);
  const [isGroupNameAdded, setIsGroupNameAdded] = useState<boolean>(true);
  const initialOptionsLength = 2;
  const initialIndexAdder = 1;
  const optionsLimit = 26;

  return (
    <div className="flex flex-col bg-grey-bg p-4 rounded-md">
      <div className="flex justify-between pb-2 mb-2">
        {!isGroupNameAdded ? (
          <>
            <input
              type="text"
              placeholder="Group name"
              className="focus:outline-none pr-1 bg-accent-shd7 text-sm text-dark max-w-12xl w-full"
              value={groupName}
              onChange={(e): void => {
                setGroupName(e.target.value);
              }}
            />

            <div className="flex">
              <button
                type="button"
                className="h-4 w-4 bg-error-shd7 focus:outline-none rounded-full flex justify-center items-center mr-xs"
                onClick={deleteGroupHandler}
              >
                <XIcon className="fill-error" />
              </button>
              <button
                type="button"
                className="h-4 w-4 bg-success-shd7 focus:outline-none rounded-full flex justify-center items-center"
                onClick={(): void => {
                  setIsGroupNameAdded(true);
                }}
              >
                <CheckSmall className="fill-success" />
              </button>
            </div>
          </>
        ) : (
          <>
            <span className="text-dark-grey font-normal text-sm">
              {groupName}
            </span>
            <ThreeDotsIcon
              className="cursor-pointer"
              onClick={(): void => {
                setIsGroupNameAdded(false);
              }}
            />
          </>
        )}
      </div>
      {options.map((option, index) => (
        <div className="mb-2" key={option.id}>
          <TextDefault
            id={`${option.id}`}
            letter={option.letter}
            deletable={options.length > initialOptionsLength}
            deleteInputHandler={(): void => {
              deleteOptionHandler(option.id);
            }}
            placeholder={`Option ${index + initialIndexAdder}`}
          />
        </div>
      ))}
      {options.length < optionsLimit ? (
        <div className="py-2.5">
          <button
            type="button"
            className="text-accent cursor-pointer flex items-center self-start focus:outline-none"
            onClick={(): void => {
              addOptionHandler(groupId);
            }}
          >
            <PlusCircle className="fill-accent mr-1 my-3xxs" />
            <span className="text-sm font-medium">Add choice</span>
          </button>
        </div>
      ) : null}
    </div>
  );
};

export default OptionGroup;
