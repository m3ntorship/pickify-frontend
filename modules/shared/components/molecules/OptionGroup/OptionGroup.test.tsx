import React from 'react';
import { render, screen } from '@testing-library/react';
import OptionGroup from './OptionGroup';
import * as ETextInput from '../../atoms/TextInputs/types/ETextInput';

describe('OptionGroup molecule', () => {
  it('should render the exact options being passed', () => {
    const options = [
      { id: '0', value: '' },
      { id: '1', value: '' },
    ];
    render(
      <OptionGroup
        variantMessage={(): ETextInput.Variants => {
          return ETextInput.Variants.Default;
        }}
        options={options}
        groupId="0"
      />,
    );
    const two = 2;
    expect(screen.getByTestId('optionsWrapper').childElementCount).toBe(two);
  });

  it('should not render the add options button if more than 26 options being passed', () => {
    const options = [];
    const zero = 0;
    const one = 1;
    const optionsLimit = 26;
    for (let i = zero; i < optionsLimit; i + one) {
      options.push({ id: `${i}`, value: '' });
    }
    render(
      <OptionGroup
        variantMessage={(): ETextInput.Variants => {
          return ETextInput.Variants.Default;
        }}
        options={options}
        groupId="0"
      />,
    );
    expect(screen.queryByText('Add choice')).toBeNull();
  });
});
