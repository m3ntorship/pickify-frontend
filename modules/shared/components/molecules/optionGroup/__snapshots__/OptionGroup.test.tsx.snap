// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionGroup component should render () OptionGroup when we apply () 1`] = `
<div
  className="options-wrapper"
>
  <div>
    <h3
      className="post-caption"
    >
      nalyzing Delaware Frozen
    </h3>
  </div>
  <div>
    <div
      className="container-for-text-poll"
      data-testid="option"
    >
      <button
        className="btnBody"
        data-testid="eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53d"
        disabled={false}
        id="eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53d"
        onClick={[Function]}
        type="button"
      >
        <span
          className="flex-container"
        >
          <span
            className="letter"
          >
            a
          </span>
          <p
            className="option"
          >
             
            ali tramsy
             
          </p>
          <span />
        </span>
      </button>
    </div>
    <div
      className="container-for-text-poll"
      data-testid="option"
    >
      <button
        className="btnBody"
        data-testid="eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53c"
        disabled={false}
        id="eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53c"
        onClick={[Function]}
        type="button"
      >
        <span
          className="flex-container"
        >
          <span
            className="letter"
          >
            b
          </span>
          <p
            className="option"
          >
             
             ahmed essam
             
          </p>
          <span />
        </span>
      </button>
    </div>
    <div
      className="container-for-text-poll"
      data-testid="option"
    >
      <button
        className="btnBody"
        data-testid="eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53c"
        disabled={false}
        id="eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53c"
        onClick={[Function]}
        type="button"
      >
        <span
          className="flex-container"
        >
          <span
            className="letter"
          >
            c
          </span>
          <p
            className="option"
          >
             
             omar gamal
             
          </p>
          <span />
        </span>
      </button>
    </div>
  </div>
</div>
`;
